common/src/main/java/org/opensearch/ml/common/transport/connector/MLCreateConnectorInput.java

```diff
@@ -148,7 +148,14 @@ public static MLCreateConnectorInput parse(XContentParser parser, boolean update
                     parameters = getParameterMap(parser.map());
                     break;
                 case CONNECTOR_CREDENTIAL_FIELD:
-                    credential = parser.mapStrings();
+                    // We need to filter out any key string that is trying to imitate the subfield of the secretArn of the credential map
+                    credential = new HashMap<>();
+                    Map<String, String> credentialKeyToAdd = parser.mapStrings();
+                    for (String key : credentialKeyToAdd.keySet()) {
+                        if (!key.startsWith("secretArn.")) {
```

Comment:
for Opensource we don't care about `secretArn.`, right? Please evaluate again if this code should be here or not? Or how can we make it generic?

Reply:
The reason we submit this PR in github instead of gitfarm directly is to avoid future autosync conflict. As our AOS specific code accumulates we are receiving more and more sev2 TT on this issue, so before we are working out a clear SOP I would like to keep the github/gitfarm repo as sync as possible.

Reply:
All ARN subfield forbid